### ISSUE TITLE
Difference between SummingMergeTree and AggregatingMergeTree

### ISSUE THREAD

**Issue by kris-swann** [ClickHouse/clickhouse-docs#975]
Hello, not sure if this is the best place to ask this, so please feel free to point me somewhere else if it isn't.

For the life of me I can't seem to find a good explanation of what the effective difference between SummingMergeTree and AggregatingMergeTree are

As best I can tell from the docs and all the discourse I've seen, SummingMergeTree is basically just a wrapper around AggregatingMergeTree that offers slightly better syntax if you are only doing sum aggregations? (i.e. Any SummingMergeTree can be expressed as an equivalent AggregatingMergeTree?)

Are there any performance/storage differences between the two table engines or is it exclusively syntactic differences?

I found this old PR that seems to suggest they are equivalent? https://github.com/ClickHouse/ClickHouse/issues/3866

---

I think if they are equivalent, it would be incredibly helpful to call that out in the docs.

I think if they are not equivalent and there are actual performance/storage differences, it would be incredibly helpful to call that out in the docs as well.

Basically I'm asking for a direct comparison between the two.

TYSM, let me know if any other info would be helpful!

**Comment by rfraposa** [comment]
`SummingMergeTree` is just an extension of `AggregatingMergeTree` where the numeric column types are summed together. So you don't have to use `sumSimpleState` or `sumState` for columns that you want summed together.

What is maybe not obvious is that `SummingMergeTree` can have columns that are not summed - but instead use any of the `-State` functions. In other words, `SummingMergeTree` can have any aggregate columns that `AggregatingMergeTree` can have, with the convenience of not having to do anything extra to columns you want summed.

Hope that helps...it's a good question.

**Comment by kris-swann** [comment]
Thank you, yes that does answer my question!

**Comment by nomikz** [comment]
I am also stuggling a lot with understanding the difference between summing and aggregatinge mergeTree engines.
So the only benefit of summing over aggregating is less syntax for fields that need to be summed?
So for SummingMergeTree:
- I can omit SimpleAggregateFunction(sum, numeric_data_type) for destination table 
- I can omit sumSimpleState() in materialized view with just sum()
- I can omit sumMerge() and just use sum()
Are those points true for SummingMergeTree?

Docs says [The Summation in the Aggregatefunction Columns](https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/summingmergetree#the-summation-in-the-aggregatefunction-columns):

> For columns of AggregateFunction type ClickHouse behaves as AggregatingMergeTree engine aggregating according to the function.

I can user summing and aggregating mergeTrees interchangeable as long as I alwasy indicate SimpleAggregateFunction, sumSimpleState, sumMerge, can't I?

**Comment by rfraposa** [comment]
> So the only benefit of summing over aggregating is less syntax for fields that need to be summed?

It's not the "only" benefit, but your logic is correct. Think of `SummingMergeTree` as being equal to `AggregatingMergeTree` except there is a default aggregation - namely `sum` - that gets applied to numeric columns. 

> can I use summing and aggregating mergeTrees interchangeable as long as I alwasy indicate SimpleAggregateFunction, sumSimpleState, sumMerge?

You don't need to indicate `SimpleAggregateFunction`, etc. If a column is a numeric data type, its values will be added together for rows with the same sort key (during the collapsing of rows).
